diff --git a/JSS/.ipynb_checkpoints/paper_results-checkpoint.py b/JSS/.ipynb_checkpoints/paper_results-checkpoint.py
index c41682f..96c4835 100755
--- a/JSS/.ipynb_checkpoints/paper_results-checkpoint.py
+++ b/JSS/.ipynb_checkpoints/paper_results-checkpoint.py
@@ -99,11 +99,12 @@ def train_func():
         "simple_optimizer": False,
         "_fake_gpus": False,
     }
-
-    wandb.init(config=default_config, group='10Minutes')
+    wandb.init(config=default_config, group='10Minutes', project='PPOJss')
     ray.init()
 
     config = wandb.config
+    
+    print(config['test'])
 
     ModelCatalog.register_custom_model("fc_masked_model_tf", FCMaskedActionsModelTF)
 
diff --git a/JSS/paper_results.py b/JSS/paper_results.py
index 099fd74..96c4835 100755
--- a/JSS/paper_results.py
+++ b/JSS/paper_results.py
@@ -99,11 +99,12 @@ def train_func():
         "simple_optimizer": False,
         "_fake_gpus": False,
     }
-
     wandb.init(config=default_config, group='10Minutes', project='PPOJss')
     ray.init()
 
     config = wandb.config
+    
+    print(config['test'])
 
     ModelCatalog.register_custom_model("fc_masked_model_tf", FCMaskedActionsModelTF)
 
