diff --git a/JSS/.ipynb_checkpoints/paper_results-checkpoint.py b/JSS/.ipynb_checkpoints/paper_results-checkpoint.py
index fe23e69..0163456 100755
--- a/JSS/.ipynb_checkpoints/paper_results-checkpoint.py
+++ b/JSS/.ipynb_checkpoints/paper_results-checkpoint.py
@@ -130,7 +130,7 @@ def train_func():
     config.pop('lr_end', None)
 
     stop = {
-        "time_total_s": 1000,
+        "time_total_s":  3 * 60 * 60,
     }
 
     start_time = time.time()
diff --git a/JSS/env/__pycache__/JSS.cpython-37.pyc b/JSS/env/__pycache__/JSS.cpython-37.pyc
index b896a7e..b25c728 100644
Binary files a/JSS/env/__pycache__/JSS.cpython-37.pyc and b/JSS/env/__pycache__/JSS.cpython-37.pyc differ
diff --git a/JSS/paper_results.py b/JSS/paper_results.py
index fe23e69..0163456 100755
--- a/JSS/paper_results.py
+++ b/JSS/paper_results.py
@@ -130,7 +130,7 @@ def train_func():
     config.pop('lr_end', None)
 
     stop = {
-        "time_total_s": 1000,
+        "time_total_s":  3 * 60 * 60,
     }
 
     start_time = time.time()
