diff --git a/JSS/main.py b/JSS/main.py
index 002296e..a2c0c59 100644
--- a/JSS/main.py
+++ b/JSS/main.py
@@ -16,6 +16,7 @@ from ray.tune import CLIReporter, register_env
 from ray.tune.logger import DEFAULT_LOGGERS
 from ray.tune.integration.wandb import WandbLogger
 
+
 from JSS.models import FCMaskedActionsModel
 
 pio.orca.config.use_xvfb = True
@@ -29,16 +30,16 @@ register_env("jss_env", env_creator)
 
 
 if __name__ == "__main__":
+    os.environ["WANDB_API_KEY"] = '3487a01956bf67cc7882bca2a38f70c8c95f8463'
     ray.init()
     torch, nn = try_import_torch()
     ModelCatalog.register_custom_model("fc_masked_model", FCMaskedActionsModel)
-    os.environ["WANDB_API_KEY"] = '3487a01956bf67cc7882bca2a38f70c8c95f8463'
     config = ppo.DEFAULT_CONFIG.copy()
     fcnet_architectures = [[512, 512], [1024, 1024], [2048, 2048]]
     config['seed'] = 0
     config['framework'] = 'torch'
     config['env'] = 'jss_env'
-    config['env_config'] = {'instance_path': '/home/local/IWAS/pierre/PycharmProjects/JSS/JSS/env/instances/ta51'}
+    config['env_config'] = {'instance_path': '/home/jovyan/JSS/JSS/env/instances/ta51'}
     config['model'] = {
         "fcnet_activation": tune.grid_search(["relu", "tanh"]),
         "custom_model": "fc_masked_model",
@@ -55,14 +56,12 @@ if __name__ == "__main__":
     config['num_envs_per_worker'] = tune.grid_search([1, 2])
     config['callbacks'] = CustomCallbacks
     config['num_workers'] = 7
-    '''
     config["logger_config"] = {
         'wandb': {
             'project': 'RLLibJSS',
             'api_key': '3487a01956bf67cc7882bca2a38f70c8c95f8463',
         }
     }
-    '''
     stop = {
         "time_total_s": 600,
     }
@@ -71,7 +70,8 @@ if __name__ == "__main__":
     reporter.add_metric_column("custom_metric_time_step_min")
     analysis = tune.run(PPOTrainer, config=config, stop=stop, progress_reporter=reporter,
                         fail_fast=True,
-                        checkpoint_at_end=True)
+                        checkpoint_at_end=True,
+                        loggers=[WandbLogger])
     best_trained_config = analysis.get_best_config(metric="episode_reward_max")
     print("Best config: ", best_trained_config)
     save_config = open("{}_{}.json".format(time.time(), config['env_config']['instance_path']), "w+")
