diff --git a/JSS/.ipynb_checkpoints/sweep_ppo-checkpoint.yaml b/JSS/.ipynb_checkpoints/sweep_ppo-checkpoint.yaml
index 7bceb26..868a304 100644
--- a/JSS/.ipynb_checkpoints/sweep_ppo-checkpoint.yaml
+++ b/JSS/.ipynb_checkpoints/sweep_ppo-checkpoint.yaml
@@ -12,7 +12,7 @@ parameters:
     max: 32000
   num_envs_per_worker:
     min: 1
-    max: 4
+    max: 8
   rollout_fragment_length:
     min: 256
     max: 1256
@@ -21,7 +21,7 @@ parameters:
     max: 1000
   lr:
     min: 0.000001
-    max: 0.005
+    max: 0.001
   entropy_coeff:
     min: 0.00000001
     max: 0.01
diff --git a/JSS/.ipynb_checkpoints/train_wandb-checkpoint.py b/JSS/.ipynb_checkpoints/train_wandb-checkpoint.py
index 50d92b4..ee182b1 100644
--- a/JSS/.ipynb_checkpoints/train_wandb-checkpoint.py
+++ b/JSS/.ipynb_checkpoints/train_wandb-checkpoint.py
@@ -110,6 +110,7 @@ def train_func():
         "fcnet_activation": "relu",
         "custom_model": "fc_masked_model_tf",
         'fcnet_hiddens': [config['layer_size'] for k in range(config['layer_nb'])],
+        "vf_share_layers": False,
     }
     config['env_config'] = {
         'instance_path': config['instance_path']
diff --git a/JSS/sweep_ppo.yaml b/JSS/sweep_ppo.yaml
index 7bceb26..868a304 100644
--- a/JSS/sweep_ppo.yaml
+++ b/JSS/sweep_ppo.yaml
@@ -12,7 +12,7 @@ parameters:
     max: 32000
   num_envs_per_worker:
     min: 1
-    max: 4
+    max: 8
   rollout_fragment_length:
     min: 256
     max: 1256
@@ -21,7 +21,7 @@ parameters:
     max: 1000
   lr:
     min: 0.000001
-    max: 0.005
+    max: 0.001
   entropy_coeff:
     min: 0.00000001
     max: 0.01
diff --git a/JSS/train_wandb.py b/JSS/train_wandb.py
index 50d92b4..ee182b1 100644
--- a/JSS/train_wandb.py
+++ b/JSS/train_wandb.py
@@ -110,6 +110,7 @@ def train_func():
         "fcnet_activation": "relu",
         "custom_model": "fc_masked_model_tf",
         'fcnet_hiddens': [config['layer_size'] for k in range(config['layer_nb'])],
+        "vf_share_layers": False,
     }
     config['env_config'] = {
         'instance_path': config['instance_path']
